sympy/solvers/ode/riccati.py

```diff
@@ -0,0 +1,506 @@
+            sol = list(linsolve([sums], [c[i]], x))
+            if len(sol):
+                c[i] = sol[0][0]
+    for i in range(len(c)):
+        if len(S(c[i]).atoms(Dummy)):
+            c = c[:i]
+            break
+    return c[::-1] if reverse else c
+
+
+def compute_degree(x, poles, choice, c, d, N):
+    ybar = 0
+    m = Poly(d[choice[0]][-1], x, extension=True)
+
+    # Calculate the first (nested) summation for ybar
+    # as given in Step 9 of the Thesis (Pg 82)
+    for i in range(len(poles)):
+        for j in range(len(c[i][choice[i + 1]])):
+            # If one of the poles is infinity and its coefficient is
+            # not zero, the given solution is invalid as there will be
+            # a c/(x - oo)^j term in ybar for some c and j
+            if poles[i] != oo:
+                # return m, ybar, ybar, ybar, False
+                ybar += c[i][choice[i + 1]][j]/(x - poles[i])**(j+1)
+        m -= Poly(c[i][choice[i + 1]][0], x, extension=True)
+
+    # Calculate the second summation for ybar
+    for i in range(N+1):
+        ybar += d[choice[0]][i]*x**i
+    numy, deny = [Poly(e, x, extension=True) for e in ybar.together().as_numer_denom()]
+    return m.expr, ybar, numy, deny, True
+
+
+def solve_aux_eq(numa, dena, numy, deny, x, m):
+    # Assume that the solution is of the type
+    # p(x) = C0 + C1*x + ... + Cm*x**m
+    psyms = symbols(f'C0:{m}', cls=Dummy)
+    domain = ZZ[psyms]
+    psyms += (1, )
+    psol = 0
+    for i in range(m + 1):
+        psol += Poly(psyms[i]*x**i, x, domain=domain)
+
+    # Eq (5.16) in Thesis - Pg 81
+    auxeq = (dena*(numy.diff(x)*deny - numy*deny.diff(x) + numy**2) - numa*deny**2)*psol
+    if m >= 1:
+        px = psol.diff(x)
+        auxeq += 2*px*numy*deny*dena
+    if m >= 2:
+        auxeq += px.diff(x)*deny**2*dena
+    if m != 0:
+        # m is a non-zero integer. Find the constant terms using undetermined coefficients
+        return psol, linsolve_dict(auxeq.all_coeffs(), psyms, x), True
+    else:
+        # m == 0 . Check if 1 (x**0) is a solution to the auxiliary equation
+        return S(1), auxeq, auxeq == 0
+
+
+def solve_riccati(fx, x, b0, b1, b2):
+    # Step 1 : Convert to Normal Form
+    a = -b0*b2 + b1**2/4 - b1.diff(x)/2 + 3*b2.diff(x)**2/(4*b2**2) + b1*b2.diff(x)/(2*b2) - b2.diff(x, 2)/(2*b2)
+    a_t = a.together()
+    num, den = [Poly(e, x, extension=True) for e in a_t.as_numer_denom()]
+    num, den = num.cancel(den, include=True)
+
+    # Step 2
+    presol = []
+
+    # Step 3 : a(x) is 0
+    if num == 0:
+        presol.append(1/(x + Dummy('C1')))
+
+    # Step 4 : a(x) is a non-zero constant
+    elif x not in num.free_symbols.union(den.free_symbols):
+        presol.extend([sqrt(a), -sqrt(a)])
+
+    # Step 5 : Find poles and valuation at infinity
+    poles = find_poles(num, den, x)
+    # If a(x) has a pole at infinity, get its multiplicity
+    inf_mul = poles.get(oo, 0)
+    # Remove oo from the set of finite poles
+    poles.pop(oo, None)
```

Comment:
I actually need the multiplicity of `oo` when it is a pole since that is used in the `rational_laurent_series` function. Maybe can I just find the multiplicity inside that function.

Reply:
I've removed `oo` as pole.